lib/bap_disasm/bap_disasm_reconstructor.ml

```diff
@@ -19,51 +19,45 @@ type reconstructor = t
 let create f = Reconstructor f
 let run (Reconstructor f) = f
 
-let find_calls name roots cfg =
-  let starts = Addr.Table.create () in
-  List.iter roots ~f:(fun addr ->
-      Hashtbl.set starts ~key:addr ~data:(name addr));
-  Cfg.nodes cfg |> Seq.iter ~f:(fun blk ->
-      let () =
-        if Seq.is_empty (Cfg.Node.inputs blk cfg) then
-          let addr = Block.addr blk in
-          Hashtbl.set starts ~key:addr ~data:(name addr) in
-      let term = Block.terminator blk in
-      if Insn.(is call) term then
-        Seq.iter (Cfg.Node.outputs blk cfg)
-          ~f:(fun e ->
-              if Cfg.Edge.label e <> `Fall then
-                let w = Block.addr (Cfg.Edge.dst e) in
-                Hashtbl.set starts ~key:w ~data:(name w)));
-  starts
+let callees_of_block cfg roots blk =
```

Comment:
a bad choice of name, since you're adding the block itself to the resulting set it is no longer a callees_of_block. Suggestion, use `entries_of_block` since you're effectively collecting the entry nodes. 

Also, instead of doing lots of set unions you can pass the set along and use `Set.add`. It is more efficient.